text/0000-codify-infrastructure.md

```diff
@@ -0,0 +1,392 @@
+
+### What exactly _is_ the datafruits stack? 🤔
+
+Basically, to run on kubernetes, you need to make all of your applications (like, code that executes and runs things) [containerizable](https://opensource.com/resources/what-docker), and you need to keep your data elsewhere. We'll get into why this is the case later, but for now I'll go into these talking about the current way datafruits runs things and how we'll fit them into the new model.
+
+As near as I can tell, this is an overview of the entire tech stack that goes into making datafruits tick. I'm suuuuper shaky on this, so feedback is highly appreciated -- I'm almost certainly missing things 🙃
+
+#### Client Site (datafruits.fm) 🍉
+
+The most important (arguably) part of the whole thing is the website. The repo lives at [datafruits/datafruits](https://github.com/datafruits/datafruits). It's built on [ember.js](https://emberjs.com/), and importantly with [ember-cli-fastboot](https://github.com/ember-fastboot/ember-cli-fastboot). This means that the web server is a [super lightweight Node.js process](https://github.com/ember-fastboot/fastboot-app-server#ember-fastboot-app-server), which is trivial to containerize. More on containerization later.
+
+#### Admin Site (streampusher.com) 🚂
+
+This is the place you go where if you're Tony, you can look at all the DJs and their information, or if you're a DJ you can manage your own stuff and upload sets and things like that. I'm not _super_ sure how it works, because I'm not a DJ and I don't have access to it, and I've never been invested enough to really dig through the code.
+
+That said, it has its entire own codebase, which lives in [streampusher/frontend](https://github.com/streampusher/frontend). This, too, is an Ember app, which does NOT use ember-cli-fastboot. It currently lives on [netlify](https://www.netlify.com/), which is like a [serverless](https://aws.amazon.com/serverless/) [static site](https://en.wikipedia.org/wiki/Static_web_page) hosting platform, kind of like [GitHub Pages](https://pages.github.com/). I think. I've never used netlify, although I certainly have seen people sing its praises.
+
+At least, I think it lives on netfliy. There's a netlify config in the [`ember-cli-build.js`](https://github.com/streampusher/frontend/blob/master/ember-cli-build.js). Either way, it should be similarly easy to containerize this app the same way we'd containerize the client site.
+
+#### Static Assets 🗂
+
+Another important thing for UX is the static assets -- who doesn't love the pulsating 𝕕𝕒𝕥𝕒𝕗𝕣𝕦𝕚𝕥𝕤 logo? Currently...I'm not sure how these are served. Some of them are checked in to the client site repo, but some of them aren't. With this RFC, I propose we move all of the static assets to dedicated object hosting (like [DigitalOcean Spaces](https://www.digitalocean.com/products/spaces/)). [More on object storage](https://blog.westerndigital.com/why-object-storage/)
```

Comment:
I would love to move all the podcast mp3s to spaces. :tada: 

Also get a real CDN going. I'm not sure what our options are there. I've heard good things about [bunny cdn](https://bunnycdn.com/).

Reply:
DigitalOcean has a [built-in CDN](https://www.digitalocean.com/products/spaces/#testContentBlockSection) for their delivery :)

Also, if your DNS is with [Cloudflare](https://www.cloudflare.com/learning/cdn/what-is-caching/), they have built-in caching too.

Btw, if you're not with Cloudflare, i highly recommend them. They recently started offering zero-markup domain names, and their free plan is super generous.